File: src/main/java/study/codingtest/leetcode/dp/LC_53.kt
fun maxSubArray(nums: IntArray): Int {
  var curSum: Int = 0;
  var maxSum: Int = Int.MIN_VALUE;
  for (num in nums) {
    curSum = Math.max(num, curSum + num);
    maxSum = Math.max(maxSum, curSum);
  }
  return maxSum;
}

File: src/main/java/study/codingtest/leetcode/dp/LC_53.ts
const maxSubArray = (nums: number[]): number => {
  let curSum: number = 0;
  let maxSum: number = Number.MIN_SAFE_INTEGER;
  for (const num of nums) {
    curSum = Math.max(num, curSum + num);
    maxSum = Math.max(maxSum, curSum);
  }
  return maxSum;
};
